qiling/core.py

```diff
@@ -181,6 +183,10 @@ def __init__(
         if self.baremetal:
             self._hw = select_component('hw', 'hw')(self)
 
+        # load r2 extension
+        if enable_r2:
+            self.r2 = R2(self)
```

Comment:
I think integrating r2 into Qiling at this point is a **very bad idea**.
This should stay as extention, and by definition extention cannot be an integral part of the core - but only an optional / experimental feature.

Reply:
I agree that it shouldn't be an option in the constructor but instead I hope to make it used everywhere, with the higher level information obtained from r2.

Reply:
Yeah in this PR I also want to keep r2 as an extension without any modification of qiling core.